File: mobilka/Slovar/app/src/main/java/com/example/zametki/MainActivity.java
package com.example.zametki;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText txtKey;
    private EditText txtValue;
    private DB db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txtKey = findViewById(R.id.txt_key);
        txtValue = findViewById(R.id.txt_value);
        db = new DB(this);
    }

    public void onInsertClick(View view) {
        String key = txtKey.getText().toString().trim();
        String value = txtValue.getText().toString();

        if (TextUtils.isEmpty(key)) {
            showErrorDialog(R.string.error_title, R.string.error_empty_key);
            return;
        }
        if (db.keyExists(key)) {
            showErrorDialog(R.string.error_title, R.string.error_duplicate_key, key);
            return;
        }

        db.insert(key, value);
    }

    public void onSelectClick(View view) {
        String key = txtKey.getText().toString().trim();

        if (TextUtils.isEmpty(key)) {
            showErrorDialog(R.string.error_title, R.string.error_empty_key);
            return;
        }

        String value = db.select(key);
        if (value == null) {
            showErrorDialog(R.string.error_title, R.string.error_key_not_found, key);
            return;
        }

        txtValue.setText(value);
    }

    public void onUpdateClick(View view) {
        String key = txtKey.getText().toString().trim();
        String value = txtValue.getText().toString();

        if (TextUtils.isEmpty(key)) {
            showErrorDialog(R.string.error_title, R.string.error_empty_key);
            return;
        }
        if (!db.keyExists(key)) {
            showErrorDialog(R.string.error_title, R.string.error_key_not_found, key);
            return;
        }

        db.update(key, value);
    }

    public void onDeleteClick(View view) {
        String key = txtKey.getText().toString().trim();

        if (TextUtils.isEmpty(key)) {
            showErrorDialog(R.string.error_title, R.string.error_empty_key);
            return;
        }
        if (!db.keyExists(key)) {
            showErrorDialog(R.string.error_title, R.string.error_key_not_found, key);
            return;
        }

        new AlertDialog.Builder(this)
                .setTitle(R.string.delete_title)
                .setIcon(R.drawable.dialog_icon)
                .setMessage(getString(R.string.delete_message, key))
                .setPositiveButton(R.string.dialog_ok, (dialog, which) -> {
                    db.delete(key);
                    txtValue.setText("");
                })
                .setNegativeButton(R.string.dialog_cancel, null)
                .show();
    }

    private void showErrorDialog(int titleResId, int messageResId) {
        new AlertDialog.Builder(this)
                .setTitle(titleResId)
                .setIcon(R.drawable.dialog_icon)
                .setMessage(messageResId)
                .setPositiveButton(R.string.dialog_ok, null)
                .show();
    }

    private void showErrorDialog(int titleResId, int messageResId, String key) {
        new AlertDialog.Builder(this)
                .setTitle(titleResId)
                .setIcon(R.drawable.dialog_icon)
                .setMessage(getString(messageResId, key))
                .setPositiveButton(R.string.dialog_ok, null)
                .show();
    }
}

File: mobilka/ElementsList/app/src/main/java/com/example/elementslist/MainActivity.kt
package com.example.elementslist

import android.os.Bundle
import android.view.View
import android.view.ViewParent
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import android.widget.Spinner
import android.widget.SpinnerAdapter
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    private lateinit var lvTodo: ListView
    private val items = mutableListOf<String>()
    private lateinit var adapter: ArrayAdapter<String>
    private var selectedIndex = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        lvTodo = findViewById(R.id.lvTodo)
        adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, items)
        lvTodo.adapter = adapter

        lvTodo.setOnItemClickListener { _, _, position, _ ->
            selectedIndex = position
        }

        findViewById<Button>(R.id.btnAdd).setOnClickListener { addItem() }
        findViewById<Button>(R.id.btnEdit).setOnClickListener { editItem() }
        findViewById<Button>(R.id.btnDelete).setOnClickListener { deleteItem() }
    }

    private fun addItem() {
        val input = EditText(this)
        AlertDialog.Builder(this)
            .setTitle("Добавить задачу")
            .setView(input)
            .setPositiveButton("OK") { _, _ ->
                val text = input.text.toString()
                if (text.isNotEmpty()) {
                    items.add(text)
                    adapter.notifyDataSetChanged()
                }
            }
            .setNegativeButton("Отмена", null)
            .show()
    }

    private fun editItem() {
        if (selectedIndex == -1) return
        val input = EditText(this)
        input.setText(items[selectedIndex])
        AlertDialog.Builder(this)
            .setTitle("Изменить задачу")
            .setView(input)
            .setPositiveButton("OK") { _, _ ->
                val text = input.text.toString()
                if (text.isNotEmpty()) {
                    items[selectedIndex] = text
                    adapter.notifyDataSetChanged()
                }
            }
            .setNegativeButton("Отмена", null)
            .show()
    }

    private fun deleteItem() {
        if (selectedIndex == -1) return
        AlertDialog.Builder(this)
            .setTitle("Удаление")
            .setIcon(R.drawable.iconka)
            .setMessage("Удалить '${items[selectedIndex]}'?")
            .setPositiveButton("OK") { _, _ ->
                items.removeAt(selectedIndex)
                adapter.notifyDataSetChanged()
                selectedIndex = -1
            }
            .setNegativeButton("Отмена", null)
            .show()
    }
}